sections/design/basics-of-react.tex

```diff
@@ -0,0 +1,97 @@
+\section{Introduction to React}
+React is a framework for JavaScript that we will be using to build the frontend of the system. 
+In this section we will look at the different building blocks and concepts of React.
+React will often be used with a syntax extension to JavaScript called JSX which makes it possible to do a special kind of variable declaration.
+
+\begin{center}
+    \texttt{const element = <h1>Hello, world!</h1>;}
+\end{center}
+This variable declaration can be called a React element which can then be rendered to the DOM
+Other variables can be declared and used inside JSX by wrapping it in brackets.
+\begin{center}
+    \texttt{ReactDOM.render(element, document.getElementById('root'));}
+\end{center}
+We can render the element to the DOM by using \texttt{ReactDom.render()}. The element is passed to this function as well as the DOM node we would like it to be rendered to. In this case we are rendering the react element \texttt{element} to the DOM node \texttt{root}.
+\begin{lstlisting}
+const name = 'Josh Perez';
+const element = <h1>Hello, {name}</h1>;
+
+ReactDOM.render(
+  element,
+  document.getElementById('root')
+);
+\end{lstlisting}
+The variable that is embedded in the JSX can contain any valid JavaScript expression.
+\\\\
+In React we use components, which is a single file that holds all of the logic, html and styling that we need.
+
+\begin{lstlisting}
+import React from "react";
+    
+function App() {
+    return (
+        <div>
+            <h1>Hello React</h1>
+        </div>
+    );
+}
+
+export default App
+\end{lstlisting}
+In this example we create the component \texttt{App} and export it. By exporting it we can use the \texttt{App} component elsewhere and render it. 
+In the function we return JSX which React will then compile to html.
+The idea is to build the UI from components by exporting them and then importing them into other larger components to contain them. This means that the components act as the building blocks of the UI because we can reuse them in other components.
```

Comment:
```suggestion
The idea is to build the UI from components by exporting them and then importing them into other larger components to contain them.
This means that the components act as the building blocks of the UI because we can reuse them in other components.
```